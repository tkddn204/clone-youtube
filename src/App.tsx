import React, { useEffect } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";

function App() {
  useEffect(() => {

  }, []);

  return (
    <Router>
      <Switch>
        <Route path="/">
        </Route>
      </Switch>
    </Router>
  );
}

export default App;
