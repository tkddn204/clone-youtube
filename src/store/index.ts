import {configureStore, combineReducers, ThunkAction, Action} from '@reduxjs/toolkit';
import {connectRouter, routerMiddleware} from 'connected-react-router';
import {createBrowserHistory} from 'history';
import logger from 'redux-logger';
import countryCodeReducer from './features/country-code-slice';
import fetchVideoReducer from './features/fetch-video-slice';
import channelResultReducer from './features/channel-result-slice';
import drawerReducer from "./features/drawer-slice";

export const history = createBrowserHistory();

// Store
const rootReducer = (history: any) => combineReducers({
  router: connectRouter(history),
  countryCode: countryCodeReducer,
  searchResult: fetchVideoReducer.searchResultReducer,
  popularResult: fetchVideoReducer.popularResultReducer,
  channelResult: channelResultReducer,
  drawerState: drawerReducer
});
export const store = configureStore({
  reducer: rootReducer(history),
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware()
      .prepend(
        routerMiddleware(history)
      ).concat(logger)
});
export const getAppState = store.getState;

export type AppDispatch = typeof store.dispatch;

// Thunk
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType,
  RootState,
  unknown,
  Action<string>>;
