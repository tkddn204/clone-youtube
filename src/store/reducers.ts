import screenSizeReducer from "./features/screen-size-slice";
import countryCodeReducer from "./features/country-code-slice";
import drawerReducer from "./features/drawer-slice";
import fetchVideoReducer from "./features/fetch-video-slice";
import channelResultReducer from "./features/channel-result-slice";

export default {
  screenSize: screenSizeReducer,
  countryCode: countryCodeReducer,
  drawerState: drawerReducer,
  searchResult: fetchVideoReducer.searchResultReducer,
  popularResult: fetchVideoReducer.popularResultReducer,
  channelResult: channelResultReducer
};