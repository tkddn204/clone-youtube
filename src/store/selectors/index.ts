import {createSelector} from "@reduxjs/toolkit";
import {Youtube} from "../api/youtube";

const getVideoResult = (state: any) => state.videoResult;
const getSearchResult = (state: any) => state.searchResult;
const getPopularResult = (state: any) => state.popularResult;
const getChannelResult = (state: any) => state.channelResult;
const getDrawerState = (state: any) => state.drawerState;

export interface NormalizedYoutubeVideoInfo {
  id: string,
  title: string,
  description: string,
  channelName: string,
  publishedAt: Date,
  thumbnailSrc?: string,
  channelThumbnailSrc?: string,
  viewCount?: string,
  tags?: ReadonlyArray<string>,
  isStreaming: boolean
}

const normalize = (result: Youtube.Response.VideoItem): NormalizedYoutubeVideoInfo => ({
  id: result.id,
  thumbnailSrc: result.snippet.thumbnails.medium ? result.snippet.thumbnails.medium.url : result.snippet.thumbnails.default.url,
  title: result.snippet.title,
  description: result.snippet.description,
  channelName: result.snippet.channelTitle,
  channelThumbnailSrc: result.snippet.channelThumbnails?.default.url,
  viewCount: result.statistics?.viewCount,
  publishedAt: new Date(result.snippet.publishedAt),
  tags: result.snippet.tags,
  isStreaming: false
});

const resultNormalize = (result: any) => result.videos.map(normalize);

export const VideoSelector = createSelector(
  [getVideoResult],
  resultNormalize
)

export const SearchSelector = createSelector(
  [getSearchResult],
  resultNormalize
);

export const PopularSelector = createSelector(
  [getPopularResult],
  resultNormalize
);

export const ChannelSelector = createSelector(
  [getChannelResult],
  channelResult => channelResult.channel
);

export const VideoListSelector = createSelector(
  [SearchSelector, PopularSelector],
  (search, popular) => ({
    search, popular
  })
);

export const DrawerSelector = createSelector(
  [getDrawerState],
  drawerState => drawerState
);