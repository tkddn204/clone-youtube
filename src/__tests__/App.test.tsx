import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../store';
import App from '../App';
import {createBrowserHistory} from 'history';

describe('<App />', () => {
  test('match snapshot', () => {
    const history = createBrowserHistory();
    const {container} = render(
      <Provider store={store}>
        <App history={history}/>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});