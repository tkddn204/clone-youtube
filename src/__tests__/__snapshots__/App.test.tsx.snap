// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> match snapshot 1`] = `<div />`;
